external/fv3fit/tests/test_packer.py

```diff
@@ -188,6 +189,49 @@ def test_sklearn_unpack(dataset: xr.Dataset):
     xr.testing.assert_allclose(unpacked_dataset, dataset)
 
 
+def test_sklearn_pack_unpack_with_clipping(dataset: xr.Dataset):
+    name = list(dataset.data_vars)[0]
+    if FEATURE_DIM in dataset[name].dims:
+        pack_config = PackerConfig({name: {FEATURE_DIM: SliceConfig(3, None)}})
+        packed_array, feature_index = pack(dataset, [SAMPLE_DIM], pack_config)
+        unpacked_dataset = unpack(packed_array, [SAMPLE_DIM], feature_index)
+        expected = {}
+        for k in dataset:
+            da = dataset[k].copy(deep=True)
+            indices = pack_config.clip.get(k, {})
+            for dim, slice_config in indices.items():
+                da = da.isel({dim: slice_config.slice})
+            expected[k] = da
+        xr.testing.assert_allclose(unpacked_dataset, xr.Dataset(expected))
+
+
+def test_clip(dataset: xr.Dataset):
+    name = list(dataset.data_vars)[0]
+    if FEATURE_DIM in dataset[name].dims:
+        indices = {name: {FEATURE_DIM: SliceConfig(4, 8)}}
+        clipped_data = clip(dataset, indices)
+        expected_da = dataset[name]
+        for dim, slice_config in indices[name].items():
+            expected_da = expected_da.isel({dim: slice_config.slice})
```

Comment:
This change would let the test pick up errors in `slice_config.slice` implementation.
```suggestion
            expected_da = expected_da.isel({dim: slice(4, 8)})
```

Reply:
True. Although that is tested elsewhere already.